app/workers/kyc/kycaid/document_worker.rb

```diff
@@ -0,0 +1,77 @@
+require 'sidekiq'
+
+module KYC
+  module Kycaid
+    class DocumentWorker
+      include Sidekiq::Worker
+      KYCAID_DOC_TYPES = { "Passport" => 'PASSPORT', 'Identity card' => 'GOVERNMENT_ID', 'Driver license' => 'DRIVERS_LICENSE' }.freeze
+
+      def perform(user_id, identificator)
+        # user, that will be verified
+        @user = User.find(user_id)
+        # exact batch of the docs to be verified
+        docs = @user.documents.where(identificator: identificator)
+        # user should already have applicant on his last profile to be able to proceed documents verification
+        @applicant_id = @user.profiles.last.applicant_id
+
+        # creates a KYCAID::File for front/back and KYCAID::Document with linking files
+        front_file = docs.find_by(doc_category: 'front_side')
+        back_file = docs.find_by(doc_category: 'back_side')
+        selfie_file = docs.find_by(doc_category: 'selfie')
+
+        document = ::KYCAID::Document.create(document_params(front_file, back_file))
+        selfie_document = ::KYCAID::Document.create(selfie_image_params(selfie_file))
+
+        if document.error || document.errors || selfie_document.error || selfie_document.errors
+          Rails.logger.info("Error in document creation for: #{@user.uid}: #{document.error} #{document.errors} \
+                             or in  selfie image creation for: #{@user.uid}: #{selfie_document.error} #{selfie_document.errors}")
+          @user.labels.find_by(key: :document, scope: :private).update(value: 'rejected')
+        elsif document.document_id && selfie_document.document_id
+          docs.update_all(metadata: { document_id: @document_id }.to_json)
```

Comment:
we can use `document.document_id` here. This info is optional, only in log reasons